packages/protocol/contracts/identity/AttestationSubsidy.sol

```diff
@@ -0,0 +1,135 @@
+ * @title Contract that implements a meta-tx based cUSD subsidy for attestations
+ */
+contract AttestationSubsidy is
+  IAttestationSubsidy,
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ReentrancyGuard,
+  UsingPrecompiles
+{
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  event AttestationSubsidised(address indexed account, uint256 value);
+
+  /**
+     * @notice Used in place of the constructor to allow the contract to be upgradable via proxy.
+     * @param registryAddress The address of the registry core smart contract.
+     */
+  function initialize(address registryAddress) external initializer {
+    setRegistry(registryAddress);
+    _transferOwnership(msg.sender);
+  }
+
+  /**
+     * @notice Used to batch three operations:
+     * - Approve cUSD transfer between the beneficiary meta-wallet and the attestations contract
+     * - Transfer the cUSD subsidy to the meta-wallet
+     * - Execute the attestation request to spend the cUSD subsidy
+     * @param beneficiaryMetaWallet - the address of the beneficiary meta-wallet
+     * @param identifier - the identifier to request attestations for (see Attestations.sol)
+     * @param attestationsRequested - the number of attestations requested (see Attestations.sol)
+     * @param v,r,s - array of signatures:
+     *    -- [0] = signatures for cUSD.approve(Attestations.sol, attestationsRequested*fee)
+     *    -- [1] = signatures for Attestations.request(identifier, attestationsRequested, cUSD)
+     */
+  function requestAttestationsWithSubsidy(
+    address beneficiaryMetaWallet,
+    bytes32 identifier,
+    uint256 attestationsRequested,
+    uint8[] calldata v,
+    bytes32[] calldata r,
+    bytes32[] calldata s
+  ) external onlyOwner nonReentrant {
+    require(v.length == 2, "two signatures are required (approve,request)");
+    require(r.length == 2, "two signatures are required (approve,request)");
+    require(s.length == 2, "two signatures are required (approve,request)");
+
+    uint256 totalFee = _calculateTotalFee(attestationsRequested);
+    IMetaTransactionWallet metaWallet = IMetaTransactionWallet(beneficiaryMetaWallet);
+    IERC20 cUSD = IERC20(address(getStableToken()));
+
+    _metaApproveCUSD(metaWallet, totalFee, v[0], r[0], s[0]);
+    cUSD.transfer(beneficiaryMetaWallet, totalFee);
+    _metaRequestAttestations(metaWallet, identifier, attestationsRequested, v[1], r[1], s[1]);
+
+    emit AttestationSubsidised(beneficiaryMetaWallet, totalFee);
+  }
+
+  /**
+     * @notice executes the cUSD approval from the metaWallet to the Attestations contract
+     * @param metaWallet - the wallet in question
+     * @param totalFee - the total amount of cUSD to be approved
+     * @param v,r,s - the signature for the meta-transaction
+     */
+  function _metaApproveCUSD(
+    IMetaTransactionWallet metaWallet,
+    uint256 totalFee,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) internal {
+    bytes memory callData = abi.encodeWithSignature(
```

Comment:
Yeah we could save on gas with this we construct the calldata off-chain, but then we have two options:
1. Also provide the `attesatationRequests` & `identifier` as an argument in order to use them for extra checks
2. Extract `attesatationRequests` & `identifier` from the calldata encoded off-chain 

If we do (1) we have two bits of data which we _think_ should be connected but aren't necessarily. We would treat the calldata as if it has the same `attestationRequests` value encoded in it as the one passed as arguments, but it would be an implicit assumption.

If we do (2) it will probably end up costing the same to extract variables from the encoded calldata. Well actually the assembly `mload` thing If probably cheaper the constructing on-chain.